src/client/Microsoft.Identity.Client/Platforms/Android/AuthenticationContinuationHelper.cs

```diff
@@ -34,20 +35,25 @@ public static void SetAuthenticationContinuationEventArgs(int requestCode, Resul
             logger.Info(string.Format(CultureInfo.InvariantCulture, "Received Activity Result({0})", (int)resultCode));
 
             AuthorizationResult authorizationResult;
-            if (data.Action != null && data.Action.Equals("ReturnFromEmbeddedWebview", StringComparison.OrdinalIgnoreCase))
+
+            if (data != null && data.Action != null && data.Action.Equals("ReturnFromEmbeddedWebview", StringComparison.OrdinalIgnoreCase))
             {
                 authorizationResult = ProcessFromEmbeddedWebview(requestCode, resultCode, data);
             }
-            else if (!String.IsNullOrEmpty(data.GetStringExtra(BrokerConstants.BrokerResultV2)) || requestCode == BrokerConstants.BrokerRequestId) 
+            else if (data != null && (!String.IsNullOrEmpty(data.GetStringExtra(BrokerConstants.BrokerResultV2)) || requestCode == BrokerConstants.BrokerRequestId)) 
                 //The BrokerRequestId is an ID that is attached to the activity launch during brokered authentication
                 // that indicates that the response returned to this class is for the broker.
             {
                 AndroidBroker.SetBrokerResult(data, (int)resultCode);
                 return;
             }
-            else
+            else if (data != null || AndroidConstants.AuthCodeReceived != (int)resultCode)
             {
                 authorizationResult = ProcessFromSystemWebview(requestCode, resultCode, data);
+            } else
```

Comment:
Please have the else on a separate line.

Reply:
Java habit :)